morepath/request.py

```diff
@@ -191,12 +199,12 @@ class (if this app doesn't handle them) to another app.
         if app is SAME_APP:
             app = self.app
 
-        info = app._get_deferred_mounted_path(obj)
+        info, app = app._get_deferred_mounted_path(obj)
```

Comment:
You are right, that should be changed.

In a way `SAME_APP` is a bit of a misnomer here. What I think the `app` argument offers is a choice between a default automatism (just select the right app) and a manual override (I know which app I want). Because we no longer really care about it being the same app or not, since we look it up and as a result might end up with the same or another app.

Therefore I would suggest to move the `SAME_APP` override further down:

```python
info, app = app._get_deferred_mounted_path(obj)

if app is not SAME_APP:
    app = self.app
```

Note that unlike before we only override the app if it has been given explicitly (the condition has been inverted). This is why it would technically be nicer if `SAME_APP` was called `AUTO` in this instance. Though we do use it in other contexts, so I would not change this at the moment.

What do you think?